editor/src/components/editor/insert-callbacks.ts

```diff
@@ -134,32 +175,146 @@ function useEnterDrawToInsertForElement(elementFactory: (newUID: string) => JSXE
   )
 }
 
-export function useToInsert(): (convertTo: InsertMenuItem | null) => void {
+export function useToInsert(): (elementToInsert: InsertMenuItem | null) => void {
   const dispatch = useDispatch()
+  const builtInDependenciesRef = useRefEditorState((store) => store.builtInDependencies)
   const selectedViewsRef = useRefEditorState((store) => store.editor.selectedViews)
   const jsxMetadataRef = useRefEditorState((store) => store.editor.jsxMetadata)
+  const allElementPropsRef = useRefEditorState((store) => store.editor.allElementProps)
   const elementPathTreeRef = useRefEditorState((store) => store.editor.elementPathTree)
   const projectContentsRef = useRefEditorState((store) => store.editor.projectContents)
+  const nodeModulesRef = useRefEditorState((store) => store.editor.nodeModules)
 
   return React.useCallback(
-    (convertToMenuItem: InsertMenuItem | null) => {
-      if (convertToMenuItem != null) {
-        // Pretend floating menu state as it may be forced by the UI.
-        const floatingMenuState = floatingInsertMenuStateInsert(null, null)
-        const convertTo = convertToMenuItem.value
-        const actions = getActionsToApplyChange(
-          projectContentsRef.current,
-          jsxMetadataRef.current,
-          elementPathTreeRef.current,
-          selectedViewsRef.current,
-          floatingMenuState,
-          false,
-          false,
-          convertTo,
-        )
-        dispatch(actions, 'everyone')
+    (elementToInsert: InsertMenuItem | null) => {
+      if (elementToInsert == null) {
+        return
+      }
+      const targetParent: ElementPath | null = safeIndex(selectedViewsRef.current, 0) ?? null
+
+      if (targetParent == null) {
+        return
```

Comment:
This should probably show a toast, otherwise it fails silently in legitimate cases (i.e. nothing is selected and I click the insert menu)